src/pint/toa.py

```diff
@@ -1728,11 +1736,14 @@ def write_TOA_file(self, filename, name="unk", format="tempo2"):
             obs_obj = Observatory.get(obs)
 
             flags = flags.copy()
-            if "clkcorr" in flags.keys():
+            if "clkcorr" in flags:
                 toatime_out = toatime - time.TimeDelta(flags["clkcorr"])
             else:
                 toatime_out = toatime
-            out_str = format_toa_line(
+            out_str = (
+                "C " if isinstance(commentflag, str) and (commentflag in flags) else ""
```

Comment:
None is hashable, so you can just leave this as `commentflag in flags` and rely on the keys all being actual flags.

Reply:
What I'm doing is stricter than that.  I'm requiring that `commentflag` must be a string.  If it is anything else, including a number, and list, an array, or a `None`, then it won't write a comment.